README.ConfigureResourceReferences.md

```diff
@@ -0,0 +1,185 @@
+The common cases of such values are:
+
+*   Resource ID of another GCP resource
+*   Relative resource name of another GCP resource
+*   Self link of another GCP resource
+*   A service-generated value of another GCP resource
+
+## Identify which fields should be turned into reference fields
+
+You might already look at the
+[Terraform resource documentation](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs)
+when researching the resource. You can also take a look at the API documentation
+if needed.
+
+For example, to learn the fields in PubSubTopic, you will need to go through the
+fields under Arguments Reference section in
+[google_pubsub_topic](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/pubsub_topic),
+and you may need check the
+[REST API doc](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.topics)
+and the
+[tutorial doc](https://cloud.google.com/pubsub/docs/create-topic#managing_topics).
+
+After you have a basic understanding of the fields for the target resource, some
+approaches would help you identify whether the field is mapping to a GCP
+resource.
+
+1.  In the GCP tutorial for the target resource, if another GCP resource is
+    required before creating the target resource, and if the value of a field in
+    the other GCP resource is needed when creating the target resource, then we
+    know the field is likely a reference field. Note down the field name and the
+    other GCP resource name.
+1.  In the TF sample for the target resource, if the value of a field is an
+    attribute reference of another TF resource, then we know the field is likely
+    a reference field. Note down the field name and the other TF resource type
+    name. E.g. the `key_ring` fields mentioned in the
+    [cmek sample for PubSubTopic](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/pubsub_topic#example-usage---pubsub-topic-cmek).
+1.  Be aware of field name mentioning: `name`, `id`, `link`, `email`,
+    `service_account`, resource names under the same API of the target resource,
+    or field description mentioning `refer`, `GCP`/`google`/`cloud` resources,
+    `url`, `uri`, etc. Once you see these keywords, it's very likely that this
+    field references to gcp resource. Double-check the GCP tutorial and/or TF
+    documentation to confirm.
+
+Figure out the referenced GCP resource through reading the field description. If
+you are still unsure, contact the API team for more information. Once it's
+determined, note down the field name and the referenced GCP resource name.
+
+## Identify whether the referenced resource is TF-based or DCL-based
+
+### Turn GCP resource name or TF type name into KCC kind name
+
+Before moving on, you need to turn the referenced TF type/GCP resource names
+you've noted down into KCC kind names. Ask the KCC team for help if you still
+can't figure them out with the following instructions.
```

Comment:
"Ask the KCC team" seems not very actionable in the context of OSS. We can say file a GitHub question in this repo instead.